File: springboot-jpa/src/main/java/com/kiritor/repository/UserRepository.java
/**
 * Copyright (C), 2014-2018, xx有限公司
 * FileName: UserJpa
 */
package com.kiritor.repository;

import com.kiritor.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.io.Serializable;
import java.util.List;

/**
 * 〈〉
 *
 * @author lcore
 * @create 18/11/16
 * @since 1.0.0
 */

public interface UserRepository extends
        JpaRepository<User, Long>,
        Serializable {
    @Query("select t from User t where t.userName = ?1")
    List<User> finByUserName(String username);
}

File: springboot-kotlin/src/main/kotlin/com/kiritor/repository/UserRepository.kt
package com.kiritor.repository

import com.kiritor.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.io.Serializable


interface UserRepository: JpaRepository<User, Long>,
        Serializable {

    @Query("from User t where t.userName like %:key% or t.telPhone like %:key%")
    fun getUserByKey(@Param("key")key:String):List<User>

}